Lib/test/test_embed.py

```diff
@@ -70,15 +70,15 @@ def setUp(self):
             expecteddir = os.path.join(support.REPO_ROOT, builddir)
         else:
             exepath = os.path.join(builddir, 'Programs')
-            expecteddir = os.path.join(support.REPO_ROOT, 'Programs')
+            expecteddir = os.path.join(builddir, 'Programs')
         self.test_exe = exe = os.path.join(exepath, exename)
         if exepath != expecteddir or not os.path.exists(exe):
             self.skipTest("%r doesn't exist" % exe)
         # This is needed otherwise we get a fatal error:
         # "Py_Initialize: Unable to get the locale encoding
         # LookupError: no codec search functions registered: can't find encoding"
         self.oldcwd = os.getcwd()
-        os.chdir(support.REPO_ROOT)
+        os.chdir(builddir)
```

Comment:
I think `os.chdir(support.REPO_ROOT)` is needed for Windows.

Maybe `isfile()` failed on some machines?
https://github.com/python/cpython/blob/115b3276d07e1478c44daaa809a0e415415c5552/Lib/test/test_embed.py#L673-L682

Reply:
Or `isfile()` in `getpath.py` failed.

Reply:
Release builds fail the tests (expectedly). Debug builds pass unexpectedly because getpath.py currently misses the check of debug extension (`_d`), which will be fixed. So, separating `chdir()` is one solution.

Reply:
Removed windows.